### PR TITLE
Add configuration guidence

### PR DESCRIPTION
**Pull request by rtyley**
You can see the docs in a more human-friendly format than the PR diff here:

https://github.com/guardian/gha-scala-library-release-workflow/blob/add-configuration-guidence/README.md

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* [`sbt-release`](https://github.com/sbt/sbt-release)
+* [`sbt-sonatype`](https://github.com/xerial/sbt-sonatype) - you currently need to have this in your project's
+  `plugins.sbt`, as the [`🎊 Create artifacts`](https://github.com/guardian/gha-scala-library-release-workflow/blob/7d278d4d44e30b4b4c0f6791053bdeb40b8159cb/.github/workflows/reusable-release.yml#L141-L158)
+  job expects it, but in the future, the workflow will probably supply this automatically.
+
+### Recommended `sbt` settings
+
+[Example `build.sbt`](https://github.com/guardian/etag-caching/blob/main/build.sbt)
+
+* `scalacOptions` should include `-release:11` (also known as `-java-output-version`
```
Comment (ioannakok) [inline docs/configuration.md:42]: Should we mention that the minimum Scala version to use this option is `2.13.12`?

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+[Example `.github/workflows/release.yml`](https://github.com/guardian/etag-caching/blob/main/.github/workflows/release.yml)
+
+The functionality of `gha-scala-library-release-workflow` is provided in a
+[reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)
+called [`reusable-release.yml`](https://github.com/guardian/gha-scala-library-release-workflow/blob/main/.github/workflows/reusable-release.yml) -
+don't copy-and-paste that big file, instead just make a small `release.yml` workflow to
+[_call_](https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow)
+it (as in the example file above).
+
+Your repo will require access to [release credentials](credentials.md) to
```
Comment (ioannakok) [inline docs/configuration.md:19]: Should we mention the repo has to be tagged with the `production` label?

**Review by ioannakok — APPROVED** [review]
Looks great! Just two things you might want to also mention

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+* [`sbt-version-policy`](https://github.com/scalacenter/sbt-version-policy) - to supply automatic compatibility-based
+  version-numbering that follows [recommended Scala semver rules](https://www.scala-lang.org/blog/2021/02/16/preventing-version-conflicts-with-versionscheme.html#early-semver-and-sbt-version-policy)
+  on your library, as well as automatically setting the [`versionScheme`](https://www.scala-sbt.org/1.x/docs/Publishing.html#Version+scheme)
+  of your project to `"early-semver"` (essential).
+* [`sbt-release`](https://github.com/sbt/sbt-release)
+* [`sbt-sonatype`](https://github.com/xerial/sbt-sonatype) - you currently need to have this in your project's
+  `plugins.sbt`, as the [`🎊 Create artifacts`](https://github.com/guardian/gha-scala-library-release-workflow/blob/7d278d4d44e30b4b4c0f6791053bdeb40b8159cb/.github/workflows/reusable-release.yml#L141-L158)
+  job expects it, but in the future, the workflow will probably supply this automatically.
+
+### Recommended `sbt` settings
```
Comment (DavidLawes) [inline docs/configuration.md:38]: Not an sbt setting, but I suppose it's implicit that any project using this reusable workflow defines the next version using a `version.sbt` file (which isn't currently the case for mobile-apps-api-models). I think mapi models is an anomaly in terms of the release process though, so maybe it's superfluous to mention the necessity for a version.sbt in these docs.

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+* `releaseVersion := fromAggregatedAssessedCompatibilityWithLatestRelease().value` - to activate the
+  automatic compatibility-based version-numbering provided by the `sbt-version-policy` plugin.
+* `publish / skip := true` (rather than other legacy hacks like `publishArtifact := false`) for
+  sbt modules that don't generate artifacts (often, the 'root' project in a multi-project build). This
+  setting is respected by `sbt-version-policy` - it won't attempt to calculate compatibility on a module
+  that doesn't publish artifacts.
+* In `releaseProcess`, you'll want _fewer_ steps than
+  [the old list specified by `sbt-sonatype`](https://github.com/xerial/sbt-sonatype?tab=readme-ov-file#using-with-sbt-release-plugin),
+  now just:
+  `checkSnapshotDependencies, inquireVersions, runClean, runTest, setReleaseVersion, commitReleaseVersion, tagRelease, setNextVersion, commitNextVersion`
```
Comment (DavidLawes) [inline docs/configuration.md:55]: The `commitReleaseVersion` step requires branch protection to be dropped (although not sure if that is only an intermediate requirement while a different solution is being developed?). That may be obvious enough not to merit a comment about that here though.

**Review by DavidLawes — APPROVED** [review]
Super useful, thanks so much!

File: docs/configuration.md (hunk 1 above)
Comment (rtyley) [reply docs/configuration.md]: It's not quite _that_ new that you need 2.13.12 (see https://github.com/guardian/frontend/pull/25717 !) but yeah, good point - done in 99340564e83d511aa77134f894eed0588a2b5acf!

File: docs/configuration.md (hunk 2 above)
Comment (rtyley) [reply docs/configuration.md:28]: It's documented in https://github.com/guardian/github-secret-access (in fact, the CI tests on that repo fail and tell you if you forgot to do it!) but it does fit nicely into the new 'Repo settings' section, so yep!

File: docs/configuration.md (hunk 3 above)
Comment (rtyley) [reply docs/configuration.md:47]: Good point - added in 99340564e83d511aa77134f894eed0588a2b5acf

File: docs/configuration.md (hunk 4 above)
Comment (rtyley) [reply docs/configuration.md:75]: Good point, added in the new 'Repo settings' section in 99340564e83d511aa77134f894eed0588a2b5acf

**Pull request merged**